lms/static/js/ccx/view/ccx_schedule.js

```diff
@@ -0,0 +1,394 @@
+          );
+          this.errorMessage = this.validUnitDates(unit.start, unit.due);
+          if (!this.errorMessage) {
+            this.collection.reset(this.scheduleJson);
+            this.unSavedChanger = true;
+            this.render();
+          } else{
+            this.showErrorMessage();
+          }
+        }
+      },
+
+      hideUnitFromScheduleTree: function(chapterLocation, sequentialLocation,
+                                         verticalLocation) {
+        // hide child (can be chapter, sequential or vertical) in collection.
+        var unit = this.findUnit(
+          this.scheduleJson,
+          chapterLocation,
+          sequentialLocation,
+          verticalLocation
+        );
+
+        if (unit) {
+          this.hideUnitOnTree(
+            this.scheduleJson,
+            unit,
+            chapterLocation,
+            sequentialLocation
+          );
+          this.unSavedChanger = true;
+          this.collection.reset(this.scheduleJson);
+          this.render();
+        }
+      },
+
+      hideAllUnitFromScheduleTree: function() {
+        // clear schedule tree on remove all click
+        this.changeVisibilityOfUnitInSchedule(
+          this.scheduleJson, this.hide
+        );
+        this.unSavedChanger = true;
+        this.collection.reset(this.scheduleJson);
+        this.render();
+      },
+
+      showUnitInScheduleTree: function(
+        chapterLocation, sequentialLocation, verticalLocation,
+        startDate, dueDate
+      ) {
+        // hide child (can be chapter, sequential or vertical) in collection.
+        var units = this.findLineage(
+          this.scheduleJson,
+          chapterLocation,
+          sequentialLocation,
+          verticalLocation
+        );
+
+        if (units) {
+          this.errorMessage = this.validUnitDates(startDate, dueDate);
+          if (!this.errorMessage) {
+            units.map(this.show);
+            var unit = units[units.length - 1];
+            if (!_.isUndefined(unit)) {
+              unit.start = startDate;
+              unit.due = dueDate;
+            }
+            this.changeVisibilityOfUnitInSchedule([unit], this.show);
+            this.unSavedChanger = true;
+            this.collection.reset(this.scheduleJson);
+            this.render();
+          } else {
+            this.showErrorMessage();
+          }
+        }
+      },
+
+      showAllUnitsInScheduleTree: function() {
+        // show all units i.e chapters, subsections and verticals
+        // in schedule tree
+        this.changeVisibilityOfUnitInSchedule(
+          this.scheduleJson, this.show
+        );
+        this.unSavedChanger = true;
+        this.collection.reset(this.scheduleJson);
+        this.render();
+      },
+
+      applyUnitToTree: function(tree, unit, chapterLocation, sequentialLocation) {
+        // sync tree ith unit
+        _.each(tree, function(chapter, chapterIndex) {
+          if (chapterLocation) {
```

Comment:
Instead of nesting so deeply, just early out. It will make this all easier to read.

```js
if (!chapterLocation) {
  return;
}
```